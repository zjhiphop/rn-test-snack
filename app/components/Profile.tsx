import { observer } from "mobx-react-lite"
import * as React from "react"
import { Image, Pressable, StyleProp, TextInput, TextStyle, View, ViewStyle } from "react-native"

import { colors, typography } from "../theme"
import { Text } from "./Text"

export interface ProfileProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const Profile = observer(function Profile(props: ProfileProps) {
  const { style } = props
  const $styles = [$container, style]
  // test mocked store
  // const store = useStores()
  // console.log(store)
  const HomeIcon = require("../../assets/svg/home.svg")

  return (
    <View style={$styles}>
      <Text style={$text}>Hello</Text>
      <Image
        accessibilityLabel="welcome face"
        source={require("../../assets/images/welcome-face.png")}
      />
      <HomeIcon accessibilityLabel="home svg" fill="gray" width={100} height={100}></HomeIcon>
    </View>
  )
})

export const NewMessageForm = function NewMessageForm({ onSend }: { onSend?: any }) {
  const [inputText, setInputText] = React.useState("")

  const handleSend = () => {
    if (onSend) {
      onSend(inputText)
    }
    setInputText("")
  }

  return (
    <View>
      <TextInput placeholder="Message" value={inputText} onChangeText={setInputText} />
      <Pressable onPress={handleSend}>
        <Text>Send</Text>
      </Pressable>
    </View>
  )
}

const $container: ViewStyle = {
  justifyContent: "center",
}

const $text: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: 14,
  color: colors.palette.primary500,
}
