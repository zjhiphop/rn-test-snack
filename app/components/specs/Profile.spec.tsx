// https://reactnativetesting.io/component/testing/
// https://github.com/testing-library/jest-native
import { fireEvent, render, screen } from "@testing-library/react-native"
import React from "react"
import renderer from "react-test-renderer"
import { NewMessageForm, Profile } from "../Profile"

describe("Profile (visibility)", () => {
  beforeEach(() => {
    render(<Profile />)
  })
  it("renders", () => {
    expect(screen.getByText("Hello")).toBeTruthy()
  })

  it('display "welcome face" image', () => {
    expect(screen.getByLabelText("welcome face")).toBeVisible()
  })

  it('display "home" svg icon', () => {
    expect(screen.getByLabelText("home svg")).toBeVisible()
  })
})

describe("Message (Interactive)", () => {
  describe("clicking send", () => {
    it("clears the message field", () => {
      render(<NewMessageForm />)

      fireEvent.changeText(screen.getByPlaceholderText("Message"), "Hello world")
      fireEvent.press(screen.getByText("Send"))

      expect(screen.getByPlaceholderText("Message")).toHaveProp("value", "")
    })

    it("calls the onSend prop", () => {
      const messageText = "Hello world"
      const sendHandler = jest.fn()
      render(<NewMessageForm onSend={sendHandler} />)

      fireEvent.changeText(screen.getByPlaceholderText("Message"), "Hello world")
      fireEvent.press(screen.getByText("Send"))

      expect(sendHandler).toHaveBeenCalledWith(messageText)
    })
  })
})

describe("Renders Snapshot correctly", () => {
  const tree = renderer.create(<Profile />).toJSON()
  expect(tree).toMatchSnapshot()
})