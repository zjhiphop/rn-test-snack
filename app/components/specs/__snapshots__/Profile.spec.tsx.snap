// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Array [
      Object {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {},
        Array [
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
          Object {
            "fontFamily": "spaceGroteskRegular",
          },
          Object {
            "color": "#191015",
          },
        ],
        undefined,
        undefined,
        Object {
          "color": "#C76542",
          "fontFamily": "spaceGroteskRegular",
          "fontSize": 14,
        },
      ]
    }
  >
    Hello
  </Text>
  <RCTImageView
    accessibilityLabel="welcome face"
    internal_analyticTag={null}
    resizeMode="cover"
    source={
      Array [
        Object {
          "height": undefined,
          "uri": undefined,
          "width": undefined,
        },
      ]
    }
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
        "width": undefined,
      }
    }
  />
  <SvgMock
    accessibilityLabel="home svg"
    fill="gray"
    height={100}
    width={100}
  />
</View>
`;
